l2geth/rollup/sync_service.go

```diff
@@ -839,51 +861,58 @@ func (s *SyncService) verifyFee(tx *types.Transaction) error {
 		// If fees are not enforced and the gas price is 0, return early
 		return nil
 	}
-	// When the gas price is non zero, it must be equal to the constant
-	if tx.GasPrice().Cmp(fees.BigTxGasPrice) != 0 {
-		return fmt.Errorf("tx.gasPrice must be %d", fees.TxGasPrice)
-	}
-	l1GasPrice, err := s.RollupGpo.SuggestL1GasPrice(context.Background())
-	if err != nil {
-		return err
-	}
+
+	// Ensure that the user L2 gas price is high enough
 	l2GasPrice, err := s.RollupGpo.SuggestL2GasPrice(context.Background())
 	if err != nil {
 		return err
 	}
-	// Calculate the fee based on decoded L2 gas limit
-	gas := new(big.Int).SetUint64(tx.Gas())
-	l2GasLimit := fees.DecodeL2GasLimit(gas)
-
-	// Only count the calldata here as the overhead of the fully encoded
-	// RLP transaction is handled inside of EncodeL2GasLimit
-	expectedTxGasLimit := fees.EncodeTxGasLimit(tx.Data(), l1GasPrice, l2GasLimit, l2GasPrice)
-
-	// This should only happen if the unscaled transaction fee is greater than 18.44 ETH
-	if !expectedTxGasLimit.IsUint64() {
-		return fmt.Errorf("fee overflow: %s", expectedTxGasLimit.String())
-	}
 
-	userFee := new(big.Int).Mul(new(big.Int).SetUint64(tx.Gas()), tx.GasPrice())
-	expectedFee := new(big.Int).Mul(expectedTxGasLimit, fees.BigTxGasPrice)
+	// Reject user transactions that do not have large enough of a gas price.
+	// Allow for a buffer in case the gas price changes in between the user
+	// calling `eth_gasPrice` and submitting the transaction.
 	opts := fees.PaysEnoughOpts{
-		UserFee:       userFee,
-		ExpectedFee:   expectedFee,
+		UserFee:       tx.GasPrice(),
+		ExpectedFee:   l2GasPrice,
 		ThresholdUp:   s.feeThresholdUp,
 		ThresholdDown: s.feeThresholdDown,
 	}
+
 	// Check the error type and return the correct error message to the user
 	if err := fees.PaysEnough(&opts); err != nil {
 		if errors.Is(err, fees.ErrFeeTooLow) {
-			return fmt.Errorf("%w: %d, use at least tx.gasLimit = %d and tx.gasPrice = %d",
-				fees.ErrFeeTooLow, userFee, expectedTxGasLimit, fees.BigTxGasPrice)
+			// TODO: update these error messages
```

Comment:
Going to need to sync with @K-Ho re: the canonical error messages

Reply:
This is resolved, based on https://www.notion.so/optimismpbc/How-to-pay-Fees-in-Optimistic-Ethereum-f706f4e5b13e460fa5671af48ce9a695